utils/misc/rl_utils/rl_utils/utils/observation_collector/observation_units/unity_collector_unit.py

```diff
@@ -0,0 +1,265 @@
+        _received_ped_safe_dist (bool): Flag indicating if pedestrian safe dist state data has been 
+            received.
+        _received_obs_safe_dist (bool): Flag indicating if obstacle safe dist state data has been 
+            received.
+
+        _first_reset (bool): Flag indicating if it is the first reset.
+    """
+
+    # Retrieved information
+    _collision: bool
+    _ped_safe_dist: bool
+    _obs_safe_dist: bool
+    _image_depth: np.ndarray
+    _image_color: np.ndarray
+
+    # Subscriptions
+    _collision_sub: rospy.Subscriber
+    _ped_safe_dist_sub: rospy.Subscriber
+    _obs_safe_dist_sub: rospy.Subscriber
+    _image_depth_sub: rospy.Subscriber
+    _image_color_sub: rospy.Subscriber
+
+    # Received Flags
+    _received_collision: bool
+    _received_ped_safe_dist: bool
+    _received_obs_safe_dist: bool
+    _received_image_depth: bool
+    _received_image_color: bool
+
+    _first_reset: bool
+
+    def __init__(self, ns: Namespace, observation_manager, *args, **kwargs) -> None:
+        """
+        Initialize the UnityCollectorUnit.
+
+        Args:
+            ns (Namespace): Namespace for the collector unit.
+            observation_manager: Observation manager holding this collector unit.
+        """
+        super().__init__(ns, observation_manager)
+        
+        self._enable_rgbd = rospy.get_param("rgbd/enabled", False)
+        if self._enable_rgbd:
+            self._image_bridge = CvBridge()
+
+        self._train_mode = rospy.get_param("train_mode", True)
+        self._collision = False
+        self._ped_safe_dist = False
+        self._obs_safe_dist = False
+
+        self._collision_sub: rospy.Subscriber = None
+        self._ped_safe_dist_sub: rospy.Subscriber = None
+        self._obs_safe_dist_sub: rospy.Subscriber = None
+
+        self._received_collision = False
+        self._received_ped_safe_dist = False
+        self._received_obs_safe_dist = False
+
+        self._first_reset = True
+
+    def init_subs(self):
+        """
+        Initialize the subscribers for robot state and sensor data.
+        """
+        if self._train_mode:
+            # only use these observations in train mode since they
+            # are only used for reward calculation
+            self._collision_sub = rospy.Subscriber(
+                self._ns(TOPICS.COLLISION),
+                Collision,
+                self._cb_collision,
+                tcp_nodelay=True,
+            )
+            self._ped_safe_dist_sub = rospy.Subscriber(
+                self._ns(TOPICS.PED_SAFE_DIST),
+                Collision,
+                self._cb_ped_safe_dist,
+                tcp_nodelay=True,
+            )
+            self._obs_safe_dist_sub = rospy.Subscriber(
+                self._ns(TOPICS.OBS_SAFE_DIST),
+                Collision,
+                self._cb_obs_safe_dist,
+                tcp_nodelay=True,
+            )
+        if self._enable_rgbd:
```

Comment:
create a dedicated collector unit for rgb case, safe dist and collision (more modular)

Reply:
@tuananhroman good point. I created one specific for RGBD and one for unity-specific reward-specific obs. The latter wouldn't be used separately so I think this separation is good.